app/components/market/submit/submit-market-application.component.ts

```diff
@@ -1,208 +1,173 @@
+        let jobParameters = [];
+        let poolParameters = [];
+        if (this.jobTemplate && this.jobTemplate.parameters){
+            jobParameters = Object.keys(this.jobTemplate.parameters);
+        }
+        if (this.poolTemplate && this.poolTemplate.parameters){
+            poolParameters = Object.keys(this.poolTemplate.parameters);
+        }
         let fg = {};
-        for (let key of parameterKeys) {
+
+        for (let key of jobParameters) {
             if ("defaultValue" in this.jobTemplate.parameters[key]) {
                 const defaultValue = String(this.jobTemplate.parameters[key].defaultValue);
                 fg[key] = new FormControl(defaultValue);
             } else {
                 fg[key] = new FormControl();
             }
         }
-        this.jobFormGroup = this.formBuilder.group(fg);
-
-        this.form = this.formBuilder.group({
-            job: this.jobFormGroup,
-            pool: this.pickedPool,
-        });
-        if (this._formValue) {
-            this.form.setValue(this._formValue);
+        for (let key of poolParameters) {
+            if ("defaultValue" in this.poolTemplate.parameters[key]) {
+                const defaultValue = String(this.poolTemplate.parameters[key].defaultValue);
+                fg[key] = new FormControl(defaultValue);
+            } else {
+                fg[key] = new FormControl();
+            }
         }
-        this.form.valueChanges.debounceTime(400).distinctUntilChanged().subscribe((newFormValue) => {
-            this.router.navigate([], {
-                relativeTo: this.route,
-                queryParams: {
-                    formParams: JSON.stringify(newFormValue),
-                },
-            });
-        });
+        this.form = new FormGroup(fg);
     }
 
-    public getContainerFromFileGroup(fileGroup: string) {
-        return fileGroup && `fgrp-${fileGroup}`;
-    }
-
-    @autobind()
-    public submit() {
-        this.error = null;
-        const jobParams = this.jobFormGroup.value;
-        // RPC takes in Template JSON object and Parameter JSON object
-        const obs = this.pythonRpcService.callWithAuth("submit-ncj-job", [this._buildJobTemplate(), jobParams]);
+    private _runJobWithPool(expandedPoolTemplate, jobInputs){
+        delete expandedPoolTemplate.id;
+        this.jobTemplate.job.properties.poolInfo = {
+            autoPoolSpecification: {
+            autoPoolIdPrefix: "jobname",
+            poolLifetimeOption: "job",
+            keepAlive: false,
+            pool: expandedPoolTemplate,
+            },
+        };
+        const obs = this.pythonRpcService.callWithAuth("submit-ncj-job", [this.jobTemplate, jobInputs]);
         obs.subscribe({
-            next: (data) => this._redirectToJob(data),
+            next: (data) => this._redirectToJob(),
             error: (err) => this.error = ServerError.fromPython(err),
         });
-        return obs;
     }
 
-    private _redirectToJob(data) {
```

Comment:
As mentioned above, I think we need to pass a parameter back in here.

Regarding this API, I think passing in a 'data' object is a poor design. The method only requires the jobId, and should not need to know about the structure of some random data structure. If the data structure ever changes, then this method would need to get updated with it, which doesn't make sense. The interface should be kept as clean and simple as possible.